### PR TITLE
[TIMOB-23406] Android: add 'scrolling' event to listView 

### PR DESCRIPTION
**Pull request by m1ga**
JIRA: https://jira.appcelerator.org/browse/TIMOB-23406

Partiy PR

Based on this Code from Stackoverflow: http://stackoverflow.com/a/25564729/5193915
Using the onScroll in JAVA but only firering an event to Titanium when the direction changes. So instide Ti you won't get too many calls.

Will handle big (bigger than screen) and small listItems. Using computeVerticalScrollOffset is better than checking for   "listview.getChildAt(0).getTop()" because it will be recycled once in a while.

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java [hunk 1]
```diff
@@ -99,6 +100,25 @@
 	public static final int BUILT_IN_TEMPLATE_ITEM_TYPE = 2;
 	public static final int CUSTOM_TEMPLATE_ITEM_TYPE = 3;
 
+	public class ListViewScollEvent extends ListView {
```
Comment (hansemannn) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java]: Scoll?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java [hunk 2]
```diff
@@ -366,6 +390,23 @@ public void onScroll(AbsListView view, int firstVisibleItem, int visibleItemCoun
 			{
 				_firstVisibleItem = firstVisibleItem;
 				_visibleItemCount = visibleItemCount;
+				int scrolledOffset = listView.getVerticalScrollOffset();
+			    if (scrolledOffset != mInitialScroll) {
```
Comment (hansemannn) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java]: Indentation

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java [hunk 3]
```diff
@@ -366,6 +390,23 @@ public void onScroll(AbsListView view, int firstVisibleItem, int visibleItemCoun
...
+					if (scrolledOffset > mInitialScroll) {
+						scrollUp = 1;
+					} else {
+						scrollUp = -1;
+					}
+					if (scrollUp != newScrollUp) {
+						KrollDict eventArgs = new KrollDict();
+						eventArgs.put("direction", (scrollUp>0)?"up":"down");
```
Comment (hansemannn) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java]: `eventArgs.put("direction", (scrollUp > 0) ? "up" : "down");`

**Comment by ashcoding** [comment]
Code reviewed and functionally tested.
**Accepted**

**Pull request merged**